ScriptCompensateCrossBleeding.cs

```
@@ -1,22 +1,16 @@
-﻿using System;
+using System;
 using System.Drawing;
 using System.Threading.Tasks;
 using UVtools.Core;
 using UVtools.Core.Extensions;
 using UVtools.Core.Scripting;
-using UVtools.Core.Layers;
 using Emgu.CV;
-using Emgu.CV.Structure;
-using Emgu.CV.CvEnum;
-
-using Microsoft.Toolkit.HighPerformance;
-
 
 namespace UVtools.ScriptSample;
 
 public class ScriptCompensateCrossBleeding : ScriptGlobals
 {
-    readonly ScriptNumericalInput<int> LayerBleed = new()
+    readonly ScriptNumericalInput<ushort> LayerBleed = new()
```

Why is `ushort` better? No memory saved, nothing is cheaper. I would understand `unsigned` type, but why `short`?

---

It's a matter of sense and limits. If your value is never less than `1` and no more than `500` why not to use the type that better reflect that range? (In this case ushort), that way you ensure the following:

1. Value can never be negative. (This solves the checks for `if(var <= -1) throw new Exception`)
2. Value can never be more than ushort.MaxValue (65,535). Even if that limit is huge compared to your code Maximum, is way better error than int.MaxValue (2,147,483,647)
3. Other people who see your code will auto know that value is never to be less than 0 or more than 65,535. That is not possible with a `int` without read code and all constrains on the variable.
4. Datatypes shorter than `uint` can be assigned to `int`, eg: `int number = ushort.MaxValue; // Valid` 

`int` may perform better on operations but due last points I think it worth the "trade-off", still it's a personal taste once limits are handled properly.

---

Thanks for the opinion, I tend to disagree (note, I consider this as a friendly discussion and not that it matters here):
1. the value cannot be negative, but it underflow. With unsigned type, you cannot write a reasonable assert for underflow, and when debugging, you only get a strangely large value that doesn't make sense (but a negative value can help you understand how the underflow happened)
2. I agree that having types that precisely fit is a good thing. However, unless I can get such type without much hassle (e.g., in C++ I am used to having `RangeInt< LOW, HIGH>`) I see no point in where the limit is.
3. This is where knowing the limit of 2^16 doesn't help me understand the code. It instead rises misleading questions "Why is overflowing on 16bit int needed?"

---

> Thanks for the opinion, I tend to disagree

No problem on that at all, as told either way is ok once limits are handled. It's more a personal preference/habit in the end. 
For script inputs they are clamp internally based on your set Min/Max.

1. That's true and a pit on `unsigned` types. It should throw a exception IMO as they do for overflow.
2. C# do not have utility class such `RangeInt`, it can be made but will loss the native type to a class. The good practice is to use `Math.Camp()` on setter, and optionally on getter if required(since member can be changed internally). Also there are a nice Microsoft toolkit package ([Guard](https://learn.microsoft.com/en-us/windows/communitytoolkit/developer-tools/guard)) that simplify the range and negative checks with a line. Anyway script inputs are already clamp and no need this.

Fell free to revert to `int` and/or keep using `int` in future scripts, next time I will take your preference and not change that.
Amazing work on these scripts BTW 👍